fairlearn/metrics/_extra_metrics.py

```diff
@@ -6,38 +6,175 @@
 These are metrics which are not part of `scikit-learn`.
 """
 
+import numpy as np
 import sklearn.metrics as skm
 
 from ._balanced_root_mean_squared_error import _balanced_root_mean_squared_error  # noqa: F401
 from ._mean_predictions import mean_prediction, _mean_overprediction, _mean_underprediction  # noqa: F401,E501
 from ._selection_rate import selection_rate  # noqa: F401,E501
 
+_NEED_TWO_UNIQUE_Y_VALS = "Must have two unique y values"
+_NEED_POS_LABEL_IN_Y_VALS = "Must have pos_label in y values"
 
-def true_positive_rate(y_true, y_pred, sample_weight=None):
-    r"""Calculate the true positive rate (also called sensitivity, recall, or hit rate)."""
+
+def _get_labels_for_confusion_matrix(y_true, y_pred, pos_label):
+    r"""Figure out the labels argument for skm.confusion_matrix.
+
+    This is an internal method used by the true/false positive/negative
+    rate metrics (and hence are restricted to binary data). We compute
+    these using the confusion matrix.
+    In general, we expect the input data to be :math:`\in {0, 1}`
+    with 1 being the 'positive' label. However, users might want to have
+    a different pair of values, and then specify the positive label
+    for themselves.
+    This method prepares the `labels` argument of
+    :py:func:\sklearn.metrics.confusion_matrix` based on the
+    user's specifications.
+
+    Parameters
+    ----------
+    y_true : array-like
+        The true values for the metric computation
+
+    y_pred : array-like
+        The predicted alues for the metric computation
+
+    pos_label : scalar
+        The value in the true and predicted arrays to treat as positive
+
+    Returns
+    -------
+    list
+        A two element list, consisting of the unique values of the two arrays
+        with the positive label listed last
+    """
+    my_labels = list(np.unique(np.concatenate((y_true, y_pred), axis=None)))
+    if len(my_labels) != 2:
```

Comment:
we should only be raising if there are more than 2 labels. if there is only 1 label present, there is no ambiguity in defining TPR, FPR, TNR, FNR--simply depending on whether that label equals 'pos_label' or no.

Reply:
Yes.... I had been going back and forth on that point. My thinking had been that users might like to know that their dataset was (to coin a phrase) trivial, but I do not feel strongly on the point. I'll update.